common/define-grammar.js

```diff
@@ -471,12 +474,21 @@ module.exports = function defineGrammar(dialect) {
         )
       ),
 
-      rest_parameter: $ => seq(
+      rest_type: $ => seq(
```

Comment:
It doesn't seem right to call this rule `rest_type`, since it captures the *name* of a rest parameter, but actually *does not* contain its type. Could you take a look at this and see if there's a more intuitive structure that's possible? It may be ok to just *not* reuse any rules between Tuples and parameters.

Also, maybe I'm misinterpreting this, so feel free to explain if you think I am.

Reply:
Makes sense. I've chosen another name for it in https://github.com/tree-sitter/tree-sitter-typescript/pull/118/commits/1adf7ac06d1704f385977d63dcae3136a03da7d5.